test/index.js

```diff
@@ -698,6 +698,212 @@ describe("mongoose_delete with options: { deletedId : true, deletedIdType: Strin
+      return this.skip();
+    }
+    
+    const session = await mongoose.startSession();
+    try {
+      session.startTransaction();
+      
+      const doc = await TestModel.findOne({ name: 'Anakin Skywalker' });
+      const deletedDoc = await doc.delete({ 
+        deletedBy: userId, 
+        deletedId: actionId,
+        session: session 
+      });
+      
+      // Verify document was marked deleted within the transaction
+      deletedDoc.deleted.should.equal(true);
+      deletedDoc.deletedBy.toString().should.equal(userId.toString());
+      deletedDoc.deletedId.toString().should.equal(actionId.toString());
+      should.exist(deletedDoc.deletedAt);
+      
+      // Find the document within the transaction to verify it's deleted
+      const foundInTxn = await TestModel.findWithDeleted({ name: 'Anakin Skywalker' }, null, {session: session});
+      foundInTxn[0].deleted.should.equal(true);
+      
+      // Abort transaction - changes should be rolled back
+      await session.abortTransaction();
+      
+      // Verify document was not deleted after rollback
+      const notDeleted = await TestModel.findOne({ name: 'Anakin Skywalker' });
+      should.exist(notDeleted);
+      notDeleted.deleted.should.equal(false);
+    } catch (err) {
+      await session.abortTransaction();
+      should.not.exist(err);
+    } finally {
+      session.endSession();
+    }
+  });
+
+  it("deleteById() -> should support transactions with session parameter", async function () {
+    if (mongooseMajorVersion < 5) {
+      return this.skip();
+    }
+    
+    const session = await mongoose.startSession();
+    try {
+      session.startTransaction();
+      
+      const doc = await TestModel.findOne({ name: 'Yoda' });
+      
+      // Delete by ID with session
+      const result = await TestModel.deleteById(doc._id, { 
+        deletedBy: userId,
+        session: session
+      });
+      
+      // Verify operation was successful
+      expect(result).to.be.mongoose_ok();
+      expect(result).to.be.mongoose_count(1);
+      
+      // Commit the transaction
+      await session.commitTransaction();
+      
+      // Verify document was deleted after commit
+      const deleted = await TestModel.findOneDeleted({ name: 'Yoda' });
+      deleted.deleted.should.equal(true);
+    } catch (err) {
+      should.not.exist(err);
+    } finally {
+      session.endSession();
+    }
+  });
+
+  it("delete() with multiple documents -> should support transactions with session parameter", async function () {
+    if (mongooseMajorVersion < 5) {
+      return this.skip();
+    }
+    
+    const session = await mongoose.startSession();
+    try {
+      session.startTransaction();
+      
+      // Delete multiple documents with session
+      const result = await TestModel.delete(
+        { name: { $in: ['Anakin Skywalker', 'Obi-Wan Kenobi'] } }, 
+        { 
+          deletedBy: userId, 
+          session: session
+        }
+      );
```

Comment:
🤔 I'd do this as two separate calls, an `$in` call will be seen as just a single query and doesn't really test the use case for a transaction
```suggestion
      // Delete multiple documents with session
      const result1 = await TestModel.delete(
        { name: 'Anakin Skywalker' }, 
        { 
          deletedBy: userId, 
          session: session
        }
      );
      
       const result2 = await TestModel.delete(
        { name: 'Obi-Wan Kenobi' }, 
        { 
          deletedBy: userId, 
          session: session
        }
      );
```